### DISCUSSION TITLE
Default behavior for REPL-capable interpreters

### DISCUSSION THREAD

**Discussion by michaelb** [michaelb/sniprun#32] [Q&A]
Would you rather have to:

- opt-in (upvote)
- opt-out (downvote)
- default to REPL only when there is no no drawback/compromise, or additional requirement (heart)
?

Opt-in would make the default behavior more consistent between interpreters (most can't do REPL) , but users may miss out on the feature.

**Comment by cassepipe** [comment]
opt-out 

what is the current state ?

**Reply by michaelb** [reply]
For most languages/interpreters, the current default is opt-in

**Reply by cassepipe** [reply]
What's the rationale ? Aren't all people interested in writing repl supported languages interested in repl capabilities ?

**Reply by michaelb** [reply]
Mostly due to an old bug that created the artificial requirement of running twice the first snippet, as the first 'run' of sniprun would fork into a background instance of the actual language interpreter. At that time, dark races conditions made it not worth the effort, and I didn't have a reason to impose the unfriendly 'run twice the first snippet' by default.

That, plus sometimes the REPL interpreters have some weird behaviors, like remembering a variable from long ago that you didn't expect, combining very badly with some other features (live mode), or straight up being unreliable (interpreters would get blocked because of weird previous input)

**Reply by cassepipe** [reply]
Indeed I need to run twice for `elixir` to evaluate code. Let's hope this is only weird behaviour form `iex` :)
